### PR TITLE
ELEC-126: Build system

### PR DESCRIPTION
**Pull request by karlding**
If we're planning on doing a multi-page setup, then we probably want Webpack to generate multiple bundles at once by using the entry object (see [example](https://medium.com/@rajaraodv/webpack-the-confusing-parts-58712f8fcad9#8711)).

I've arbitrarily chosen some linting options, but feel free to mess things around if they don't make sense.

### CODE REVIEW

**Comment by mwrouse** [comment]
Just curious to why your team made the decision for this to be multiple pages. I can't imaging something like this ever needing to be multiple pages, ours isn't/hasn't/and won't be. 


Also, why are you even using node and webpack? I wouldn't think this needs to be complex enough to warrant being more complicated than something requireJs could handle. Or even having modules written by someone else. 

All I would think you would be doing is receiving (and maybe sending) data over web sockets. Any modern browser will support window. You just might have to extend the window in TypeScript to get access to `window.WebSocket`.

File: README.md [hunk 1]
```diff
@@ -13,9 +13,38 @@ The telemetry client is a web frontend that displays data pushed from the server
...
+yarn run lint:ts
+yarn run lint:scss
+```
+
+#### Build
+
+```bash
+yarn run build
+```
+
```
Comment (ckitagawa) [inline README.md]: Want to add a section about starting the prod version with oneof php, python or go? They all have some kind of simple default http server.

Also are there any special rules for adding other .ts files? Would I just import them to app.ts?

**Comment by ckitagawa** [comment]
@mwrouse thank you for asking! Neither myself nor @karlding is up-to-date on best practices regarding web development. You may have a very good point about require.js and we will look into it. Would this allow us to draw on external graphing and table libraries that we are installing as node modules? In addition to our own source files? If so it is definitely worth looking into.

I am aware of the `window.WebSocket` but we were considering using Socket.IO on top of that for the purposes of using the websocket like a PubSub for different elements to subscribe to based on message type. Is there a particular benefit of a 'raw' websocket rather than relying upon a library to add a layer of abstraction?

We are planning to have at least two pages because we will use one page to display inside our vehicle as our driver display showing speed, the state of change, power consumption and other indicators like the state of drive and turn signals. This will be backed by the same server architecture as the actual telemetry client which will have a separate page to display more detailed information in our chase vehicle.

**Comment by mwrouse** [comment]
@ckitagawa As a web developer myself who spends most of my time Javascript/TypeScript I really dislike Node and NPM. I have only used it for a school project once and it was a terrible experience. I really think it removes the burden of understanding what your code is doing and places it somewhere else. 

You may have heard about the [NPM left-pad](http://blog.npmjs.org/post/141577284765/kik-left-pad-and-npm) situation ([another article](http://www.businessinsider.com/npm-left-pad-controversy-explained-2016-3)). 
The whole thing really revealed a huge problem with the JavaScript community about having dependency heavy applications when it's not needed. How so many developers chose to use a pre-build 11-line long module instead of writing it themselves, it's crazy.

Right now your project has 42 direct dev-dependencies. 😁

One of them, as you mentioned is [plottable](https://www.npmjs.com/package/plottable) which is just a wrapper for [D3.js](https://github.com/d3/d3). And you'll also get all the dependencies for plottable, one of which is D3. And you still have to compile it (a website--compile a website--that seems so strange to me). 

I've heard D3 is pretty complicated (that's because you can do some *ridiculous* graphs with D3) so we are using [Chart.js](http://www.chartjs.org/), which is easier to use but shows what we need (really only basic graphs).

Now, Socket.io, as far as I'm aware of, is really only beneficial if you need something that you can quickly move code from one app to another. This is because it will automagically detect the type of connect you should have (web sockets, ajax, polling, etc...), I think this is based on the URL. But since you already know,  why not just use raw web sockets then. 

```javascript 
// Plain JavaScript Web Sockets 
var connection = new WebSocket("ws://localhost:1014"); 

connection.onopen = function(e) { 

    // Subscribe 
    connection.onmessage = function(event) { 
        // Message received 
    };

    // Publish
    connection.send("Meow"); 
};


// Same thing with Socket.io 
var connection = new io.Socket("localhost", { 
    port: 1014
}); 
connection.connect(); 

// Subscribe
connection.on('message', function(msg) { 
   // Message received 
});

connection.on("connect", function() { 
    // Publish
    connection.send("Meow"); 
});
```

**Pull request closed without merging**